### PR TITLE
fix(Input): fixed hidden field empty spaces

### PR DESCRIPTION
**Pull request by cstatro**
### PR Checklist

this PR makes sure our hidden fields don't create empty space inside of the gridform. I have some screenshots of the issue in the ticket below. 

- [ ] Related to designs:
- [x] Related to JIRA ticket: GROW-2350
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/Form/Input.tsx [hunk 1]
```diff
@@ -117,7 +127,11 @@ export const Input = forwardRef<HTMLInputElement, InputWrapperProps>(
...
 
     return (
-      <Box position="relative" textColor={color}>
+      <Box
+        display={type === 'hidden' ? 'none' : undefined}
+        position="relative"
+        textColor={color}
+      >
         <AsComponent
           {...rest}
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Form/Input.tsx:136]: `rest` no longer includes `type`?

File: packages/gamut/src/Form/Input.tsx [hunk 2]
```diff
@@ -117,7 +127,11 @@ export const Input = forwardRef<HTMLInputElement, InputWrapperProps>(
     const ShownIcon = Icon || icon;
 
     return (
-      <Box position="relative" textColor={color}>
+      <Box
+        display={type === 'hidden' ? 'none' : undefined}
```
Comment (j10sanders) [inline packages/gamut/src/Form/Input.tsx:131]: maybe 'inherit' instead of undefined?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]

**Review by j10sanders — APPROVED** [review]
makes sense to me! not sure about using undefined though.

File: packages/gamut/src/Form/Input.tsx (hunk 1 above)
Comment (cstatro) [reply packages/gamut/src/Form/Input.tsx:126]: whoops yep unit tests caught this , just pushed up the fix 😅

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.a34f70.0<br/>@codecademy/gamut-labs@16.0.2-alpha.a34f70.0<br/>@codecademy/gamut@32.18.5-alpha.a34f70.0<br/>@codecademy/styleguide@37.5.5-alpha.a34f70.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609c26e94d264d00dcd91a9f--gamut-preview.netlify.app>https://609c26e94d264d00dcd91a9f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609c26e94d264d00dcd91a9f>Deploy Logs</a></p>

File: packages/gamut/src/Form/Input.tsx (hunk 2 above)
Comment (cstatro) [reply packages/gamut/src/Form/Input.tsx]: my concern with doing that is that explicitly setting this to something that's normally implicit could have some unintended side effect, @JoshuaKGoldberg  you got any opinions on this?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/Form/Input.tsx]: Yeah my general inclination is to stick with `undefined` (as it is now) to keep things minimally different and not add extra CSS unless we need it.

**Review by JoshuaKGoldberg — COMMENTED** [review]
Looks reasonable to me, but @dreamwasp and @codecaaron have been working in this area more than me recently.

File: packages/gamut/src/Form/Input.tsx [hunk 3]
```diff
@@ -117,7 +117,11 @@ export const Input = forwardRef<HTMLInputElement, InputWrapperProps>(
     const ShownIcon = Icon || icon;
 
     return (
-      <Box position="relative" textColor={color}>
+      <Box
+        display={rest?.type === 'hidden' ? 'none' : undefined}
```
Comment (codecaaron) [inline packages/gamut/src/Form/Input.tsx:121]: Do bots check for visibility when filling these out? 

I generally see these honeytraps be `position: absolute; left: -10000px;` to preserve their visibility but not within the actual flow.

Comment (cstatro) [reply packages/gamut/src/Form/Input.tsx:121]: this isn't used for honey pots, just hidden inputs

**Review by codecaaron — APPROVED** [review]
LGTM

File: packages/gamut/src/Form/Input.tsx (hunk 3 above)
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Form/Input.tsx:121]: `rest` will always exist:

```suggestion
        display={rest.type === 'hidden' ? 'none' : undefined}
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Cool!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.ec94ec.0<br/>@codecademy/gamut-labs@16.0.2-alpha.ec94ec.0<br/>@codecademy/gamut@32.18.5-alpha.ec94ec.0<br/>@codecademy/styleguide@37.5.5-alpha.ec94ec.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.103-alpha.6a30dc.0<br/>@codecademy/gamut-labs@17.1.1-alpha.6a30dc.0<br/>@codecademy/gamut@32.18.8-alpha.6a30dc.0<br/>@codecademy/styleguide@38.1.1-alpha.6a30dc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a29f77a2088f00bb12a4ad--gamut-preview.netlify.app>https://60a29f77a2088f00bb12a4ad--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a29f77a2088f00bb12a4ad>Deploy Logs</a></p>

**Pull request merged**